### PR TITLE
#SB-18014 | Merging datasets from Druid Query Processor

### PR DESCRIPTION
**Pull request by kumarks1122**
@SanthoshVasabhaktula : Please review it.

### CODE REVIEW

File: python-scripts/src/main/python/dataproducts/services/consumption/dqp_merger.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        self.report_config = report_config
+        self.base_path = None
+
+
+    def drop_stale(self, report_df_, delta_df_):
+        dims = self.report_config['merge']['dims']
+        query = []
+
+        try:
+            grouped_data = delta_df_[dims].drop_duplicates()
```
Comment (SanthoshVasabhaktula) [inline python-scripts/src/main/python/dataproducts/services/consumption/dqp_merger.py]: What is this variable doing here?

File: python-scripts/src/main/python/dataproducts/services/consumption/dqp_merger.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            formatted_query = ' and '.join(["`{}`!='{}'".format(dim, delta_row[dim]) for dim in dims])
+            query.append("({})".format(formatted_query))
+
+        query = " or ".join(query)
+
+        return report_df_.query(query)
+
+
+    def rollup_report(self, report_df_):
+        rollupCol = self.report_config['rollupCol']
```
Comment (SanthoshVasabhaktula) [inline python-scripts/src/main/python/dataproducts/services/consumption/dqp_merger.py]: What is the use of rollupCol? Isn't it the same as merge.dims field?

File: python-scripts/src/main/python/dataproducts/services/consumption/dqp_merger.py (hunk 1 above)
Comment (kumarks1122) [reply python-scripts/src/main/python/dataproducts/services/consumption/dqp_merger.py]: @SanthoshVasabhaktula : There was an issue, and now it is fixed.

File: python-scripts/src/main/python/dataproducts/util/utils.py [hunk 3]
```diff
@@ -402,32 +433,53 @@ def get_data_from_blob(result_loc_, backup=False):
...
+    read a blob storage file
+    :file_path: file path of the file which will be generated by DQP
+    :param result_loc_: pathlib.Path object to store the file at.
+    :return: None
+    """
+    try:
+        local_file_path = result_loc_.joinpath(file_path)
+        os.makedirs(local_file_path.parent, exist_ok=True)
+
+        container_name = 'telemetry-data-store'
```
Comment (SanthoshVasabhaktula) [inline python-scripts/src/main/python/dataproducts/util/utils.py]: Shouldn't the container name come from config?

File: python-scripts/src/main/python/dataproducts/util/utils.py [hunk 4]
```diff
@@ -438,28 +490,25 @@ def post_data_to_blob(result_loc_, backup=False):
...
             date_name = result_loc_.parent.name
             report_name = result_loc_.parent.parent.name
-            block_blob_service.create_blob_from_path(
+            azure_utils.post_data_to_store(
                 container_name=container_name,
                 blob_name=report_name + '/' + date_name + '/' + file_name,
                 file_path=str(result_loc_)
             )
         else:
             container_name = 'reports'
```
Comment (SanthoshVasabhaktula) [inline python-scripts/src/main/python/dataproducts/util/utils.py]: Move the container name to config

**Pull request merged**